app/src/main/java/org/jellyfin/mobile/player/PlayerNotificationHelper.kt

```diff
@@ -0,0 +1,122 @@
+package org.jellyfin.mobile.player
+
+import android.app.Application
+import android.app.Notification
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.content.IntentFilter
+import android.os.Build
+import androidx.core.content.getSystemService
+import org.jellyfin.mobile.AppPreferences
+import org.jellyfin.mobile.BuildConfig
+import org.jellyfin.mobile.R
+import org.jellyfin.mobile.utils.Constants
+import org.jellyfin.mobile.utils.Constants.VIDEO_PLAYER_NOTIFICATION_ID
+import org.jellyfin.mobile.utils.createMediaNotificationChannel
+import org.koin.core.KoinComponent
+import org.koin.core.inject
+import java.util.concurrent.atomic.AtomicBoolean
+
+class PlayerNotificationHelper(private val viewModel: PlayerViewModel) : KoinComponent {
+    private val appPreferences: AppPreferences by inject()
+    private val context: Context = viewModel.getApplication<Application>()
+    private val notificationManager: NotificationManager? by lazy { context.getSystemService() }
+    private val receiverRegistered = AtomicBoolean(false)
+
+    val shouldShowNotification: Boolean
+        get() = appPreferences.exoPlayerAllowBackgroundAudio
+
+    @Suppress("DEPRECATION")
+    fun postNotification() {
+        if (!shouldShowNotification) return
+        val nm = notificationManager ?: return
+        val mediaSource = viewModel.mediaSourceManager.jellyfinMediaSource.value ?: return
+        val player = viewModel.playerOrNull ?: return
+
+        // Create notification channel
+        context.createMediaNotificationChannel(nm)
+
+        val style = Notification.MediaStyle().apply {
+            setMediaSession(viewModel.mediaSession.sessionToken)
+            //setShowActionsInCompactView(0, 1, 2) // TODO
+        }
+
+        val notification = Notification.Builder(context).apply {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                setChannelId(Constants.MEDIA_NOTIFICATION_CHANNEL_ID) // Set Notification Channel on Android O and above
+                setColorized(true)
+            } else {
+                setPriority(Notification.PRIORITY_LOW)
+            }
+            setSmallIcon(R.drawable.ic_notification)
+            setContentTitle(mediaSource.title)
+            mediaSource.artists?.let(::setContentText)
+            setStyle(style)
+            setVisibility(Notification.VISIBILITY_PUBLIC)
+            addAction(generateAction(R.drawable.ic_fast_rewind_black_32dp, "Rewind", Constants.ACTION_REWIND))
```

Comment:
Hardcoded string

Reply:
Copy-pasted from the RemotePlayerService, guess I should fix it there as well ^^

Reply:
Please do! 😄